lib/ansible/modules/cloud/amazon/elasticache.py

```diff
@@ -140,19 +142,19 @@
     state: rebooted
 
 """
-
-import sys
-import time
-
+from time import sleep
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info, boto3_conn
 try:
-    import boto
-    from boto.elasticache import connect_to_region
-    HAS_BOTO = True
+    import botocore
+    HAS_BOTO3 = True
```

Comment:
That was a quick update. Thanks for spending time on this. I'll try to rereview this today, though I might not have time till next week.

Here is an example of how it's imported (and not set in the try/except; if you want to do that you will need to try to import boto3 as well as botocore, despite the module not using it directly. It's not necessary to do it like I'm suggesting obviously because there are different ways to achieve the same thing.)
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_asg.py#L381-L386
And then here is how it gets checked, right after the module is created and before any boto3 calls are made:
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_asg.py#L1257-L1258